packages/ra-core/src/controller/field/useReferenceManyFieldController.ts

```diff
@@ -162,6 +162,7 @@ export const useReferenceManyFieldController = <
             filter: filterValues,
         },
         {
+            enabled: !!get(record, source),
```

Comment:
```suggestion
            enabled: get(record, source) != null,
```

Remember that `0` is a valid id

Reply:
Good catch :)